elements/rh-table/rh-table.ts

```diff
@@ -138,6 +143,19 @@ export class RhTable extends LitElement {
     if (this.#table && this.#summary) {
       this.#table.setAttribute('aria-describedby', 'summary');
     }
+
+    /* Auto-assign `data-label` attributes based on column headers */
+    if (this.#table?.tHead && this.#colHeaders) {
+      this.#rows.forEach((row, index) => {
+        const cells = row.querySelectorAll(':is(td,th)');
+        cells.forEach((cell, index) => {
+          if (!cell.hasAttribute('data-label')) {
+            const colHeader = this.#colHeaders[index].innerText;
+            cell.setAttribute('data-label', colHeader);
+          }
+        });
+      });
```

Comment:
```suggestion
    if (this.#table?.tHead && this.#colHeaders && this.#rows) {
      for (const row of this.#rows) {
        if (!row) {
          continue;
        }

        const cells = row.querySelectorAll(':is(td, th)');
        let headerIndex = 0;

        for (const cell of cells) {
          if (!cell.hasAttribute('data-label')) {
            const colHeader = this.#colHeaders[headerIndex]?.innerText;
            cell.setAttribute('data-label', colHeader);
          }

          headerIndex++;
        }
      }
    }
```

This cleans up these typescript warnings but there are still questions here.

1. This change alters the users HTML / lightdom. IIRC this may have problems associated with it (perhaps other folks can elaborate on that more).
2. If a user has a table with a `rowspan` or `colspan`, this returns `undefined`.
3. This focuses on table headers that are on the top row. As you mentioned, what about table headers in the left column?

Overall, this is probably more complicated than we may be expecting. Perhaps it could be done in specific cases (equal number of rows and columns, etc)? Probably some discussion to be had here.